### DISCUSSION TITLE
Can I make a web app that does marp-cli html,pdf,pptx conversion on the client side?

### DISCUSSION THREAD

**Discussion by link2004** [marp-team/marp#526] [Q&A] [answered]
If so, which technology should I use?

**Comment by yhatt** [comment, marked as answer]
Marp CLI has been designed for server side JavaScript runtimes such as Node.js, and you cannot use CLI in the client-side JavaScript (on the web browser). 

You can use [Marp Core](https://github.com/marp-team/marp-core) to convert Markdown into HTML on the client JS. Probably you may have to bundle the package to target for web browsers, by using JavaScript bundler (e.g. Webpack, Vite, etc).

Implementing perfect conversion for PDF, PPTX, and other formats on only client JavaScript is extremely difficult, because Marp CLI is depending on [Puppeteer ](https://pptr.dev/)(provides API _for controlling browser process itself_).

**Reply by link2004** [reply]
Thank you for your response. I was trying to implement it on the client side, but knowing that it would be technically difficult saved me from unnecessary ef
